text/0074/0074-entity-slicing.md

```diff
@@ -0,0 +1,566 @@
+## SimplifiedEntityLoader API
+
+Having the entity manifest enables different data loading strategies that are less precise, but easy to implement.
+The easiest is the `SimpleEntityLoader`  api shown here. Users need only write a function for loading entities via their IDs, and Cedar handles determining which entities need to be loaded.
+
+
+```
+/// Implement this trait to load entities based on
+/// the entity manifest.
+/// This entity loader is called "Simple" for two reasons:
+/// 1) First, it may cause database consistency challenges- `load_entity` is called multiple times.
+/// 2) Second, it is not precise- the entity manifest only requires some
+/// fields to be loaded, but this loader always loads all fields.
+pub trait SimpleEntityLoader {
+    /// Simple entity loaders must implement `load_entity`,
+    /// a function that loads entities based on their `EntityUID`s.
+    /// For each element of `entity_ids`, returns the corresponding
+    /// [`Entity`] in the output vector.
+    fn load_entity(&mut self, entity_ids: impl IntoIterator<Item = &EntityUID>) -> impl Iterator<Item = Entity>;
+    
+    /// Loads all the entities needed for a request
+    /// using the `load_entity` function.
+    fn load(
+        &mut self,
+        schema: &Schema,
+        entity_manifest: &EntityManifest,
+        request: &Request,
+    ) -> Result<Entities, EntityManifestError> {
+      ...
+    }
+}
+```
+
+
+
+
+## Drawbacks
+
+### Risk in unverified code
+
+Increasing the complexity of the Cedar API with entity manifests provides new places for users to make mistakes. For example, using an outdated entity manifest can result in unsound entity slices. Another example would be a user writing a buggy implementation of entity loading using the manifest.
+
+
+### Full Cedar Language Support
+
+The current analysis does not support the full Cedar language (see the **Computing Access Paths** section). With some effort, the implementation could be extended to support all of Cedar by more carefully tracking access paths through different types of Cedar operations. However, it’s unclear just how tricky this will be.
+
+Along a similar note, all Cedar changes in the future will need a corresponding change in the static analysis, making it harder to extend Cedar. For example, any operations that apply to records will need careful consideration.
+
+
+### Supporting Partial Loading of Sets
+
+As written, entity manifests in this RFC do not support loading only parts of a Cedar Set, or only some of the ancestors in the entity hierarchy. This is because sets are loaded on the leaves of the access trie, with no way to specify which elements are requested.
+
+To support this feature, we recommend that we take a constraint-based approach. Constraints would be attached to nodes in the access trie, specifying which elements of sets or the parent hierarchy are needed. Constraints form a small query language, so this would increase the complexity of the entity manifest.
```

Comment:
RFC 76 provides some additional ideas about how to support partial loading of ancestors. Do you still think the constraint-based approach is the right one?

Reply:
Good question. I'm not a big fan of the `@ancestors` annotation proposed- I think a more automatic solution like constraint annotations would be better.
As for partial evaluation, it would still make entity manifests more precise.